### ISSUE TITLE
Pooling method for e5 vs SIMLM

### ISSUE THREAD

**Issue by bharadwajymg** [microsoft/unilm#1423]
**Describe**
Model I am using SIMLM:

hi @intfloat , thanks a lot for helping out on how to fine tune e5 using SIMLM. Got couple of questions.
1. we have used intfloat/e5-large-unsupervised as a base pre trained model for fine-tuning using https://github.com/microsoft/unilm/blob/master/simlm/scripts/train_biencoder_marco.sh  on MSMARCO .  But this code don't introduce any of pooling (keep_pool=False by default) but above pretrained model is trained with avg pooling . Will this creates any issue?
2.  At the end of this if we want to evaluate the model will it be avg pool or CLS pooling?
3. If we keep_pool=True it introduces a linear layer , what is the impact of this on pooling?

Thanks a lot once again.

**Comment by intfloat** [comment]
Hi @bharadwajymg ,

Yes, you need to change the code to average pooling to use e5 initialization. The current SimLM codebase always assume using CLS pooling.

Please change the lines at https://github.com/microsoft/unilm/blob/78b3a48de27c388a0212cfee49fd6dc470c9ecb5/simlm/src/models/biencoder_model.py#L124-L125 to average pooling at https://github.com/microsoft/unilm/blob/78b3a48de27c388a0212cfee49fd6dc470c9ecb5/e5/utils.py#L48-L51

**Comment by bharadwajymg** [comment]
thanks a lot for your prompt reply , much appreciated . what is the use of "keep_pool" argument ? I mean the need for adding linear layer ? , sorry of this dumb question but just curious how this will impact performance .

**Comment by intfloat** [comment]
I think our code does not have an argument called `keep_pool`, can you give pointer to specific code pieces?

If you mean `add_pooler`, it is always `False` meaning that the embedding dimension is the same as the model's hidden state dimension.

**Comment by bharadwajymg** [comment]
sorry my bad , yes its add_pooler . oh perfect, thanks a lot :)

One observation which we found is, when we initialised the model with 'intfloat/e5-large-unsupervised' its giving better results on MSMARCO  than 'intfloat/e5-large-v2' is it because of learning rate ? do we need to decrease it further while training with intfloat/e5-large-v2.

**Comment by intfloat** [comment]
I believe this is an expected behavior. 

`e5-large-v2` is supposed to be an out-of-box embedding model without further fine-tuning. 

And `e5-large-unsupervised` is more suitable as initialization for further fine-tuning, but its performance without fine-tuning is not very competitive.

**Comment by bharadwajymg** [comment]
thanks a lot for answering all the queries
